eipw-lint/tests/lint_markdown_prevent_url.rs

```diff
@@ -0,0 +1,116 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+ use crate::lints::PreventUrlsNoBackticks;
+ use crate::reporters::Text; // Assuming you have a similar reporter in your crate
+ use crate::Linter;
+ 
+ #[tokio::test]
+ async fn url_with_backticks() {
+     let src = r#"---
+ header: value1
+ ---
+ 
+ Check this URL: `http://example.com/page?query=foo`
```

Comment:
You should write tests based on the examples in #104. This URL is very close to one of the examples I gave that _should not_ trigger the lint.

In other words, this URL should be allowed because its domain is `example.com`&mdash;one of the allowed example domains at the top of #104.